### PR TITLE
feature-details-slide

### PR DESCRIPTION
**Pull request by DaniilPopovDexola**
Add animation for details and custom arrow with animation

### CODE REVIEW

File: src/widgets/Details/lib/useAnimationDatails.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import { useRef } from 'react';
+
+export const useAnimationDatails = () => {
```
Comment (ZahorulkoAndrew) [inline src/widgets/Details/lib/useAnimationDatails.ts:3]: d**A**tails ?)

File: src/widgets/Details/lib/useAnimationDatails.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+  const detailsRef = useRef<HTMLDetailsElement | null>(null);
```
Comment (ZahorulkoAndrew) [inline src/widgets/Details/lib/useAnimationDatails.ts:5]: Maybe we need to use useState?

File: src/widgets/Details/lib/useAnimationDatails.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  };
+
+  const handleShrink = () => {
+    isClosing = true;
+
+    if (detailsRef.current && summaryRef.current) {
+      const startHeight = `${detailsRef.current.offsetHeight}px`;
+      const endHeight = `${summaryRef.current.offsetHeight}px`;
+
+      animation?.cancel();
```
Comment (ZahorulkoAndrew) [inline src/widgets/Details/lib/useAnimationDatails.ts:42]: use this in animateParams

File: src/widgets/Details/lib/useAnimationDatails.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import { useRef } from 'react';
+
+export const useAnimationDatails = () => {
+
+  const detailsRef = useRef<HTMLDetailsElement | null>(null);
+  const summaryRef = useRef<HTMLDivElement | null>(null);
+  const contentRef = useRef<HTMLDivElement | null>(null);
+
+  let animation: Animation | null = null;
```
Comment (ZahorulkoAndrew) [inline src/widgets/Details/lib/useAnimationDatails.ts:9]: const animationRef = useRef<Animation | null>(null);

File: src/widgets/Details/lib/useAnimationDatails.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        handleOpen();
+      }
+      else if (isExpanding || detailsRef.current.open) {
+        handleShrink();
+      }
+    }
+  }
+
+  const animateParams = (startHeight: string, endHeight: string) => {
+    return animation = detailsRef.current && detailsRef?.current.animate({
```
Comment (ZahorulkoAndrew) [inline src/widgets/Details/lib/useAnimationDatails.ts:27]: detailsRef.current?.animate

File: src/widgets/Details/lib/useAnimationDatails.ts
Comment (ZahorulkoAndrew) [inline src/widgets/Details/lib/useAnimationDatails.ts:1]: useCallback for functions

Comment (ZahorulkoAndrew) [inline src/widgets/Details/lib/useAnimationDatails.ts:1]: Add useEffect with clear animation after unmount component

File: src/widgets/Details/lib/useAnimationDatails.ts (hunk 3 above)
Comment (DaniilPopovDexola) [reply src/widgets/Details/lib/useAnimationDatails.ts:42]: done

File: src/widgets/Details/lib/useAnimationDatails.ts (hunk 1 above)
Comment (DaniilPopovDexola) [reply src/widgets/Details/lib/useAnimationDatails.ts:3]: done

File: src/widgets/Details/lib/useAnimationDatails.ts (hunk 5 above)
Comment (DaniilPopovDexola) [reply src/widgets/Details/lib/useAnimationDatails.ts:27]: state null or element (not undefined)

File: src/widgets/Details/lib/useAnimationDatails.ts
Comment (DaniilPopovDexola) [reply src/widgets/Details/lib/useAnimationDatails.ts:1]: done

Comment (DaniilPopovDexola) [reply src/widgets/Details/lib/useAnimationDatails.ts:1]: done

File: src/widgets/Details/lib/useAnimationDatails.ts (hunk 2 above)
Comment (DaniilPopovDexola) [reply src/widgets/Details/lib/useAnimationDatails.ts:5]: done

File: src/widgets/Details/lib/useAnimationDatails.ts (hunk 4 above)
Comment (DaniilPopovDexola) [reply src/widgets/Details/lib/useAnimationDatails.ts:9]: changed on useState

**Review by ITZeyz — APPROVED** [review]

**Pull request merged**